### PR TITLE
Fixed issue #16538: Can add 2 survey participantes with the same token with the RPC api

### PR DESCRIPTION
**Pull request by gabrieljenik**
Adding validation to the save operation.

What I have seen is that encryptSave doesn't do validation by default.
I haven't updated that, but shouldn't we? Why it was set to false?

### CODE REVIEW

**Comment by Shnoulle** [comment]
> What I have seen is that encryptSave doesn't do validation by default.
> I haven't updated that, but shouldn't we? Why it was set to false?

:+1:

**Comment by sushmanadendlals** [comment]
Needs to be tested after the merge, as per discussion with Dorin

File: application/helpers/remotecontrol/remotecontrol_handle.php [hunk 1]
```diff
@@ -1809,7 +1809,7 @@ public function add_participants($sSessionKey, $iSurveyID, $aParticipantData, $b
                 if ($bCreateToken) {
                     $token->generateToken();
                 }
-                if ($token->encryptSave()) {
+                if ($token->encryptSave(true)) {
```
Comment (olleharstedt) [inline application/helpers/remotecontrol/remotecontrol_handle.php:1812]: We need a unit-test for this, I think. It's not clear when and how validation can be used with encryption.

Comment (gabrieljenik) [reply application/helpers/remotecontrol/remotecontrol_handle.php:1812]: I agree unit-test could be a good way approach.

Comment (olleharstedt) [reply application/helpers/remotecontrol/remotecontrol_handle.php:1812]: Can you make one, in this PR, please? Ping me if you need help or instructions. There already is a test for the remote API that you can look at.

**Comment by gabrieljenik** [comment]
Uploaded unit test

File: application/helpers/remotecontrol/remotecontrol_handle.php (hunk 1 above)
Comment (olleharstedt) [reply application/helpers/remotecontrol/remotecontrol_handle.php:1812]: What you added looks good, but I was thinking of a new test in this file: https://github.com/LimeSurvey/LimeSurvey/blob/master/tests/unit/helpers/RemoteControlTest.php

For add_participants RPC call. Sorry for being vague. But, more test code is always good! :)

Comment (gabrieljenik) [reply application/helpers/remotecontrol/remotecontrol_handle.php:1812]: Added

Comment (olleharstedt) [reply application/helpers/remotecontrol/remotecontrol_handle.php:1812]: Great, will try it!

**Comment by Shnoulle** [comment]
> Also, we could enhance this is with some kind of attribute which tells if the token is encrypted or not as to not double encrpyt or not double decrypt.

oToken->token are never encrypted, no ? I think it must never been encrypted. 

For other attributes : it's not in https://github.com/LimeSurvey/LimeSurvey/blob/7dc7fcf7268ee64d8e7d4bafa6b2f8e017a51ca0/application/models/Survey.php#L662  ?

**Pull request merged**